app/Controllers/ValidateFeeController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\Controllers;
+
+use InvalidArgumentException;
+use Silex\Application;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use WMDE\Euro\Euro;
+use WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee\ValidateFeeRequest;
+use WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee\ValidateMembershipFeeUseCase;
+
+class ValidateFeeController {
+
+	private $app;
+	private $httpRequest;
+
+	public function __construct( Application $app, Request $httpRequest ) {
+		$this->app = $app;
+		$this->httpRequest = $httpRequest;
+	}
+
+	public function validateFee(): Response {
+		try {
+			$fee = $this->euroFromRequest();
+		}
+		catch ( InvalidArgumentException $ex ) {
+			return $this->newJsonErrorResponse( 'not-money' );
+		}
+
+		$request = ValidateFeeRequest::newInstance()
+			->withFee( $fee )
+			->withInterval( (int)$this->httpRequest->request->get( 'paymentIntervalInMonths', '0' ) )
+			->withApplicantType( $this->httpRequest->request->get( 'addressType', '' ) );
+
+		$response = ( new ValidateMembershipFeeUseCase() )->validate( $request );
+
+		if ( $response->isSuccessful() ) {
+			return $this->app->json( [ 'status' => 'OK' ] );
+		}
+
+		return $this->newJsonErrorResponse( 'too-low' );
```

Comment:
This relies on the implicit knowledge that the only error response is "amount too low". In the unlikely event that `ValidateFeeResult` gets another negative validation response, we will display the wrong reason here. 

It's a tradeoff between correctness and brevity. This code works and is unlikely to produce wrong output in the foreseeable future. The "safe" approach would be a `createJsonErrorResponse` which maps the the possible error codes of `ValidateFeeResult` to translation messages and throws an exception for all others. What do you think?

Reply:
Step one here was having a lookup in a map with two elements. Then one got removed. So as final step I dropped that lookup part. Which indeed leaves that not as explicit. I suppose you can wrap this thing in a guard if and then throw an exception at the end of the function. Not sure that is worth it though.